db/migrate/20150109083425_create_join_table_user_application_permissions.rb

```diff
@@ -0,0 +1,14 @@
+class CreateJoinTableUserApplicationPermissions < ActiveRecord::Migration
+  def change
+    create_table :user_application_permissions do |t|
+      t.integer :user_id, null: false
+      t.integer :application_id, null: false
+      t.integer :supported_permission_id, null: false
+      t.datetime :last_synced_at
+      t.timestamps
+    end
+
+    add_index :user_application_permissions, :user_id
+    add_index :user_application_permissions, [:user_id, :application_id]
```

Comment:
We could do with a unique index on `[:user_id, :application_id, : supported_permission_id]` to backup the uniqueness validation.